lldb/source/Target/SwiftLanguageRuntime.cpp

```diff
@@ -2335,6 +2336,36 @@ SwiftLanguageRuntime::GetRuntimeUnwindPlan(ProcessSP process_sp,
     return UnwindPlanSP();
   }
 
+  // If we're in the prologue of a function, don't provide a Swift async
+  // unwind plan.  We can be tricked by unmodified caller-registers that
+  // make this look like an async frame when this is a standard ABI function
+  // call, and the parent is the async frame.
+  // This assumes that the frame pointer register will be modified in the
+  // prologue.
+  Address pc;
+  pc.SetLoadAddress(regctx->GetPC(), &target);
+  SymbolContext sc;
+  if (pc.IsValid() && pc.CalculateSymbolContext(
+                          &sc, eSymbolContextFunction | eSymbolContextSymbol)) {
+    if (sc.function) {
+      Address func_start_addr = sc.function->GetAddressRange().GetBaseAddress();
+      AddressRange prologue_range(func_start_addr,
+                                  sc.function->GetPrologueByteSize());
+      if (prologue_range.ContainsLoadAddress(pc, &target)) {
+        return UnwindPlanSP();
+      }
+    }
+
+    if (sc.symbol) {
```

Comment:
Shouldn't this be `else if (sc.symbol)`? Or is there a reason to test function *and* symbol?

Reply:
I'll piggyback off this question and ask: Does anyone know off hand when this call to `CalculateSymbolContext` would return only one of `function`/`symbol`, but not the other?

Reply:
You're right.  Function is based on debug information; Symbol is based on the raw symbol table (here's a list of symbol names and addresses).  Symbol may have a prologue description based on the instruction profiler.  But the line table based size is going to be the more reliable one.  We use the Symbol as a backup if we couldn't find a Function.